bindings/ruby/lib/typelib/gccxml.rb

```diff
@@ -760,15 +713,74 @@ class << self
             #
             # @return [Array]
             attr_reader :gccxml_default_options
+            attr_reader :castxml_default_options
         end
         @gccxml_default_options = Shellwords.split(ENV['TYPELIB_GCCXML_DEFAULT_OPTIONS'] || '-DEIGEN_DONT_VECTORIZE')
+        @castxml_default_options = Shellwords.split(ENV['TYPELIB_CASTXML_DEFAULT_OPTIONS'] || '-DEIGEN_DONT_VECTORIZE')
+
+        #figure out the correct gccxml binary name, debian has changed this name 
+        #to gccxml.real
+        def self.gcc_binary_name
+            if !`which gccxml.real`.empty?
+                return "gccxml.real"
+            end
+            return "gccxml"
+        end
+
+        def self.use_castxml
+            if !`which castxml`.empty?
+                return "castxml"
+            end
+            return "gccxml"
+
+        end
+        # Runs castxml on the provided file and with the given options, and
+        # return the Nokogiri::XML object representing the result
+        #
+        # Raises RuntimeError if casrxml failed to run
+        def self.castxml(file, options)
+            cmdline = [use_castxml, *castxml_default_options, "--castxml-gccxml", '-x', 'c++', '-std=c++11']
```

Comment:
On 06.10.2015 10:12, marvin2k wrote:
>
> In bindings/ruby/lib/typelib/gccxml.rb 
> <https://github.com/orocos-toolchain/typelib/pull/66#discussion_r41235579>:
>
> > +            return "gccxml"
> > +        end
> > +
> > +        def self.use_castxml
> > +            if !`which castxml`.empty?
> > +                return "castxml"
> > +            end
> > +            return "gccxml"
> > +
> > +        end
> > +        # Runs castxml on the provided file and with the given options, and
> > +        # return the Nokogiri::XML object representing the result
> > +        #
> > +        # Raises RuntimeError if casrxml failed to run
> > +        def self.castxml(file, options)
> > +            cmdline = [use_castxml, *castxml_default_options, "--castxml-gccxml", '-x', 'c++', '-std=c++11']
>
> this enforced "c++11" needs to be cleaned up
>
Should be checked/setted as a ENV (maybe then set by autoproj) depending 
weather we build c++11 in the future
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/orocos-toolchain/typelib/pull/66/files#r41235579>.
>


-- 
  Dipl.-Inf. Matthias Goldhoorn
  Space and Underwater Robotic

  Universität Bremen
  FB 3 - Mathematik und Informatik
  AG Robotik
  Robert-Hooke-Straße 1
  28359 Bremen, Germany
  
  Zentrale: +49 421 178 45-6611
  
  Besuchsadresse der Nebengeschäftstelle:
  Robert-Hooke-Straße 5
  28359 Bremen, Germany
  
  Tel.:    +49 421 178 45-4193
  Empfang: +49 421 178 45-6600
  Fax:     +49 421 178 45-4150
  E-Mail:  matthias.goldhoorn@informatik.uni-bremen.de

  Weitere Informationen: http://www.informatik.uni-bremen.de/robotik